src/web/app/pages-admin/admin-search-page/admin-search-page.component.spec.ts

```diff
@@ -555,4 +617,93 @@ describe('AdminSearchPageComponent', () => {
     expect(spyStatusMessageService).toBeCalled();
   });
 
+  it('should show error message if fail to send course join email', () => {
+    const studentResult: StudentAccountSearchResult = {
```

Comment:
Maybe we can add a positive test case in some way for this endpoint?

Reply:
It seems quite tricky to test `window.location.href` without modifying non-test code, do you have any ideas/suggestions?

Reply:
Yup, I know. 

I think it will be sufficient to mock the response of `getCourseJoinEmail` and `getFeedbackSessionReminderEmail` method calls in `emailGenerationService`.

And then check service is called and values of `email.recipient`, `email.subject`, `email.content` are as expected.